src/Playwright/Core/Page.cs

```diff
@@ -337,7 +337,7 @@ public Task<IResponse> RunAndWaitForNavigationAsync(Func<Task> action, PageRunAn
             });
 
         public Task<IRequest> WaitForRequestAsync(string urlOrPredicate, PageWaitForRequestOptions options = default)
-            => InnerWaitForEventAsync(PageEvent.Request, null, e => e.Url.UrlMatches(urlOrPredicate), options?.Timeout);
+            => InnerWaitForEventAsync(PageEvent.Request, null, e => e.Url.UrlMatches(urlOrPredicate.CombineUrlWithBase(Context.BaseUrl)), options?.Timeout);
```

Comment:
I it easy to miss one of the call sites. I'd suggest that you expose `Context.UrlMatches(url,  urlOrPredicate)` that would look at its private BaseUrl and do the right thing. If we didn't do that upstream, we should have.